kitsune/messages/views.py

```diff
@@ -85,7 +86,26 @@ def outbox(request):
 
 @login_required
 def new_message(request):
-    form = MessageForm(request.POST or None, user=request.user)
+    initial_data = {}
+    if request.method == "GET":
+        # Check if 'to' parameter is in the GET request and prepare initial data
+        # We have to do this because messaging links from profiles are GET
+        # Also, we have to use the get_autocomplete_suggestions function on
+        # anything passed in to populate the 'to' field correctly when we
+        # don't know if the 'to' is a user or a group or none.
+        # If a name is passed in to 'to' that is the name of both a group
+        # and a user, both will be added if the user has group messaging
+        # permissions. The user can then remove the one they don't want.
+        # Without separating the to field into to_users and to_groups,
+        # we can't know which is which, so we do a best fit.
+        to = request.GET.get("to")
+        if to:
+            suggestions = json.loads(get_autocomplete_suggestions(request).content)
```

Comment:
This works fine, but I don't like that we're treating it like a web request -- converting to and from JSON, passing the `request` object, etc. -- when it's really just a function call. I think we should extract the core of `get_autocomplete_suggestions` into a non-web function that accepts a single name string as its argument, and returns a list of suggestions, and then use that function here and within `get_autocomplete_suggestions`.